KLSadd_insertion/src/updateChapters.py

```diff
@@ -1,110 +1,367 @@
-chapNums = []
+# mathPeople is just the name for the sections that are used, like Wilson, Racah, etc.
+chap_nums = []
 paras = []
-klsparas = []
-mathPeople = []
-newCommands = []  # used to hold the indexes of the commands
-ref9II = []  # Hold section search indexes
-ref14II = []
-ref9III = []  # Holds all indexes
-ref14III = []
+
+# klsaddparas contains the locations of things with "paragraph{" in KLSadd, these are subsections that need to be sorted
+klsaddparas = []
+math_people = []
+new_commands = []  # used to hold the indexes of the commands
+
+# ref9II holds the beginnings and ends of each chapter 9 section, provided and area to search
+ref9_2 = []
+
+# ref14II holds the beginnings and ends of each chapter 14 section, provided and area to search
+ref14_2 = []
+
+# ref9III holds all subsections in each chapter 9 section along with the what  ref9II holds
+ref9_3 = []
+
+# ref14III holds all subsections in each chapter 14 section along with the what ref14II holds
+ref14_3 = []
+
+# specref9 is the KLS indexes that go in chapter 9
 specref9 = []
+
+# specref14 is the KLS indexes that in chapter 14
 specref14 = []
-comms = ""  # holds the ACTUAL STRINGS of the commands
+
+check = []
+check2 = []
+check3 = []
+kls_list = []
+sortmatch = []
+
+w, h = 3, 9999
+sorterIIIcheck = [[0 for x in range(w)] for y in range(h)]
+
+sortmatch_2 = []
+
+test = "memesbananashgvlksrjc%/n"
+print test[-2]
 # 2/18/16 this method addresses the goal of hardcoding in the necessary packages to let the chapter files run as pdf's.
-# Currently only works with chapter 9, ask Dr. Cohl to help port your
-# chapter 14 output file into a pdf
+# Currently only works with chapter 9, ask Dr. Cohl to help port your chapter 14 output file into a pdf
+
+# 5/5/16 this method implements the Hypergeometric paragraphs into relevant sections for practicality
+# Later versions should attempt to: fix paragraphs with variations on this name like
+# Basic Hypergeometric Representation, q-Hypergeometric Representation
+
+
+def new_keywords(kls):
+    """
+    This section checks through the Addendum and identifies words that are valid keywords of sections
+    It then takes that list and removes duplicates, as well as printing the output to another file.
+
+    :param kls: The addendum that provides the words.
+    :return: Inputs results into a global list.
+    """
+
+    kls_list_chap = []
+    for i in kls:
+        if kls.index(i) > 313:
+            if "paragraph{" in i or "subsubsection*{" in i:
+                kls_list.append(i[i.find("{") + 1: len(i) - 2])
+
+    i = 0
+    while i < len(kls_list):
+        if "reference" in kls_list[i].lower() or "limit relation" in kls_list[i].lower():
+
+            del kls_list[i]
+        else:
+            i += 1
+    kls_list.append("Limit Relation")
+    for i in kls_list:
+        add = True
+        for j in kls_list_chap:
+            if i == j:
+                add = False
+        if add:
+            kls_list_chap.append(i)
```

Comment:
this is way too complicated for what this code block does, maybe rewrite it? (I don't remember if this was the change I already made)